model/sharing/upload.go

```diff
@@ -52,25 +53,30 @@ func (s *Sharing) Upload(inst *instance.Instance, errors int) error {
 	}
 
 	lastTry := errors+1 == MaxRetries
-	for i := 0; i < BatchSize; i++ {
-		if len(members) == 0 {
-			break
-		}
-		m := members[0]
-		members = members[1:]
-		more, err := s.UploadTo(inst, m, lastTry)
-		if err != nil {
-			errm = multierror.Append(errm, err)
-		}
-		if more {
-			members = append(members, m)
-		}
+	done := true
+	g, _ := errgroup.WithContext(context.Background())
+	for i := range members {
+		m := members[i]
+		g.Go(func() error {
```

Comment:
Don't we need to pass `m` as an argument to the goroutine to create a closure?

Reply:
`errgroup` doesn't allow to pass arguments to the goroutine. That's why I have written `for i := range members { m := members[i]` instead of the shorter `for _, m := range members`.  It forces `m` to have the correct range.

In the future, I hope that https://github.com/golang/go/wiki/LoopvarExperiment will make things easier.

Reply:
Thanks for the explanation.